src/main/java/gov/cabinetoffice/gap/adminbackend/services/GrantMandatoryQuestionService.java

```diff
@@ -0,0 +1,163 @@
+package gov.cabinetoffice.gap.adminbackend.services;
+
+import gov.cabinetoffice.gap.adminbackend.dtos.schemes.SchemeDTO;
+import gov.cabinetoffice.gap.adminbackend.entities.GrantMandatoryQuestions;
+import gov.cabinetoffice.gap.adminbackend.exceptions.NotFoundException;
+import gov.cabinetoffice.gap.adminbackend.exceptions.SpotlightExportException;
+import gov.cabinetoffice.gap.adminbackend.models.AdminSession;
+import gov.cabinetoffice.gap.adminbackend.repositories.GrantMandatoryQuestionRepository;
+import gov.cabinetoffice.gap.adminbackend.utils.HelperUtils;
+import gov.cabinetoffice.gap.adminbackend.utils.XlsxGenerator;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+import java.io.ByteArrayOutputStream;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.List;
+
+import static java.util.Optional.ofNullable;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class GrantMandatoryQuestionService {
```

Comment:
Not going to ask for it to be changed due to timescales but there is so much duplicated code in this class. We can surely do better than this.